### PR TITLE
don't rerun cmake if not necessary

### PR DESCRIPTION
**Pull request by anderspapitto**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -309,10 +309,7 @@ def build_libs(libs):
     if not IS_WINDOWS:
         if USE_NINJA:
             my_env["CMAKE_GENERATOR"] = '-GNinja'
-            my_env["CMAKE_INSTALL"] = 'ninja install'
-        else:
-            my_env['CMAKE_GENERATOR'] = ''
-            my_env['CMAKE_INSTALL'] = 'make install'
+            my_env["CMAKE_INSTALLER"] = 'ninja'
```
Comment (ezyang) [inline setup.py]: uhh... some of us don't have ninja installed?

**Comment by ezyang** [comment]
This PR doesn't look complete. If you're going to skip running cmake when the Makefile already exists, then you need some way of overriding this heuristic when cmake inevitably fails to recmake itself (even when it should). Is the only way to do that using `python setup.py clean`?

File: setup.py (hunk 1 above)
Comment (anderspapitto) [reply setup.py]: it's still under the `if`, and if this is unset then the layer below (build_pytorch_libs.sh) defaults to `make`

**Comment by anderspapitto** [comment]
@ezyang I'm pretty comfortable saying "If the build system has changed in such a drastic and subtle way that (1) the generated make/ninja files are no longer valid AND (2) the change is such that the generated makefile/ninja file doesn't know that it should trigger a re-cmaking, then a `setup.py clean` or `git clean` is in order."

**Comment by yf225** [comment]
@anderspapitto @ezyang Should we close this PR?

**Comment by ezyang** [comment]
Sorry about the delayed response (and I didn't see the edit, because I don't get notifications for comment edits).

So, it's not just "subtle build system changes" that require a re-cmake. If you add a new source file, and that source file is read in by cmake via a glob, you need to re-cmake; cmake is not going to auto-cmake in that case https://stackoverflow.com/questions/32411963/why-is-cmake-file-glob-evil

I'm generally in favor of two commands: one that lets the user run cmake+make, and one that just runs make (when you know you're not mucking around with file structure, just code). So your proposal in edit SGTM.

**Pull request closed without merging**